### ISSUE TITLE
Installation

### ISSUE THREAD

**Issue by dinosaw** [ivankayzer/daze#1]
This platform looks amazing! I'm running into issues on installing / configuring it though and would LOVE to be able to use it. Would you be able to provide just a quick write up on getting this set up?
- I cloned the repo
- I'm using Node v0.12 and performed an npm install
- Then when attempting to run gulp build I get this:

```
[root@server games]# gulp build
[16:17:01] Using gulpfile /home/user/domains/domain.tk/public_html/games/gulpfile.js
[16:17:01] Task 'build' is not in your gulpfile
[16:17:01] Please check the documentation for proper gulpfile formatting
```

I'm not even sure if this is the exact process I should be doing. So just any general information or guidance would be great!

**Comment by dinosaw** [comment]
Actually when I do this, it seems to complete:

```
[root@titan games]# gulp
[16:26:56] Using gulpfile /home/setsuna/domains/setsuna.tk/public_html/games/gulpfile.js
[16:26:56] Starting 'default'...
[16:26:56] Finished 'default' after 229 ms
```

But I'm at a loss as to what I do next?

**Comment by ivankayzer** [comment]
@dinosaw try using
`composer update`

Go for the latest commit from Object branch or use rewritten project from laravel branch.

**Comment by dinosaw** [comment]
Hi again @ivankayzer 

Would there be any way you could just jot down a quick overview of the installation process? I'm not sure if I did everything right, but I'm getting an error due to their not being any type of index page.

What I did:
```
- Cloned the repo in its current state (with the latest commit)
- Using nvm (npm 4.2.6) I run "npm install"
- During the npm install, I was prompted and walked through the configuration/installation for the Semantic UI.
- After the completed, it said "After completion navigate to semantic/ and run "gulp build" to build" which I did also successfully
```
At this point I don't know the next steps are to get the site running (e.g. where / when would I run composer update?). When I go to where I have the site hosted in its current state now, there's no index file to be found so I just get a Forbidden message. I created a MySQL database as well and filled out the appropriate fields in conf/database.php. I'm just really confused on the whole setup process. If you could elaborate that would be awesome! Thanks so much in advance for this cool looking tool!

**Comment by dinosaw** [comment]
Just to add on, I went ahead and performed `php composer update` which looks like it started installing dependencies, but then it got to some that I don't think installed successfully. I think it would help if I just had a process for the installation. Here is the log from running composer update: http://pastebin.com/raw/ThRKvU4H

**Comment by ivankayzer** [comment]
Hello there, @dinosaw.
Thank you for your interest in my project. Here is a little instruction how to setup it to run on your machine. 

```
git clone https://github.com/ivankayzer/daze.git
cd daze
composer install
```
Now configure mamp / xampp or whatever you use to use /public as root directory of the project 

```
cp .env.example .env
php artisan key:generate
```

Open the .env file and change the db credentials
Now go to https://market.mashape.com/igdbcom/internet-game-database and create an account
Search for IGDB -> go to 'pricing' tab and click 'subscribe'
Grab your API key in Applications->Default Application->'Get the keys' button
Paste your key into .env file (API_KEY)	

```
php artisan migrate
php artisan db:seed (this may take a few minutes)
npm install (Choose yes and automatic)
mv semantic public/semantic
cd public/semantic
gulp build
```


That's it. You can log in with: 

> unique@gmail.com
> root

**Comment by dinosaw** [comment]
Hey!

Thanks a ton for the installation run down. I would have never gotten this going. But everything was going very smoothly until I got to this part:

    php artisan db:seed

Unfortunately it's generating an error with no detailed logs or clear indication as to what the problem is:
```
[root@server daze]# php artisan db:seed
  [ErrorException]
  Trying to get property of non-object
```

If I'm not mistaken, `db:seed` is supposed to populate the database with content. I do see that `php artisan migrate` worked and created the structure without issue:

```
[root@server daze]# php artisan migrate
Migration table created successfully.
Migrated: 2014_10_12_000000_create_users_table
Migrated: 2014_10_12_100000_create_password_resets_table
Migrated: 2016_09_24_133015_create_games_table
Migrated: 2016_09_24_133030_create_collections_table
Migrated: 2016_09_24_133055_create_companies_table
Migrated: 2016_09_24_133134_create_themes_table
Migrated: 2016_09_24_133148_create_genres_table
Migrated: 2016_09_24_133205_create_game_modes_table
```

But not sure where to go next since db:seed isn't working. Any additional help would be great!

**Comment by ivankayzer** [comment]
@dinosaw check your API_KEY in .env file or use mine: gzWgbrz4vImshVhsAgcxLmENOleHp13IbkAjsnBrilR8QTgW7u

**Comment by dinosaw** [comment]
Hi again,

Your API Key seemed to have gotten me past that part. I'm curious as to why my API key wouldn't work? It's the same length as yours. Maybe I did something wrong when creating the application?

Anyway, I was able to continue along with the rest of your steps without a problem, however the final product is still broken. When I navigate to the root directory where the project is, I still get a forbidden message because it's looking for an index file. So when I go to root/public/ it loads the site, but the layout is broken - it's looking for resources in the wrong directory. I'm not sure why because I chose the correct location when it asked where my project folder was.

    http://things.ga/daze/public/

Any ideas what went wrong? Also is there a certain version of node you recommend? I'm using v4.2.6.

**Comment by ivankayzer** [comment]
@dinosaw I have the latest version of node: 6.9.1
But the problem you have is connected with semantic folder. Move it from the daze/ to daze/public.

```
mv semantic public/semantic
cd public/semantic
gulp build
```

**Comment by dinosaw** [comment]
Hi

I've actually already done that, but I did it again just to make sure and I'm still getting the same problem:

```
[root@server public]# pwd
/home/things/domains/things.ga/public_html/daze/public
[root@server public]# ls
css  favicon.ico  images  index.php  js  robots.txt  semantic  web.config
[root@server public]# cd semantic/
[root@server semantic]# gulp build
[16:57:43] Using gulpfile /home/things/domains/things.ga/public_html/daze/public/semantic/gulpfile.js
[16:57:43] Starting 'build'...
Building Semantic
[16:57:43] Starting 'build-javascript'...
Building Javascript
[16:57:43] Starting 'build-css'...
Building CSS
[16:57:43] Starting 'build-assets'...
Building assets
[16:57:43] Starting 'package compressed js'...
[16:57:43] Starting 'package uncompressed js'...
[16:57:43] Finished 'build-javascript' after 114 ms
[16:57:43] Starting 'package uncompressed css'...
[16:57:43] Starting 'package compressed css'...
[16:57:43] Finished 'build-assets' after 76 ms
[16:57:43] Finished 'package uncompressed js' after 36 ms
[16:57:43] Finished 'package uncompressed css' after 26 ms
[16:57:43] Finished 'package compressed js' after 52 ms
[16:57:43] Finished 'package compressed css' after 17 ms
[16:57:43] Finished 'build-css' after 105 ms
[16:57:43] Finished 'build' after 145 ms
```

**Comment by dinosaw** [comment]
When I run `gulp build`, is it supposed to create a `/dist/` directory? Because it's not. I only have:

```
[root@titan semantic]# ls
gulpfile.js  src  tasks
```

I only ask because the resources the site are attempting to pull are all within `semantic/dist`:

```
semantic/dist/semantic.min.css
semantic/dist/semantic.js
```

**Comment by ivankayzer** [comment]
Hello, @dinosaw! Finally, I found a solution!
You should use gulp build before moving semantic folder into public.

```
gulp build
cd ..
mv semantic public/semantic
```

**Comment by dinosaw** [comment]
Hello again, thanks for continuing to help me with this, it's much appreciated.

When running `gulp build` before moving the semantic folder, it did help me get a little further but there's still some URL / redirection issues. For example, if you go to my dev site, you'll see that the page loads fine:

    http://things.ga/public/

However shouldn't I be able to just navigate to `things.ga`? Whatever is causing this problem, is unfortunately causing all links to break as well. To reproduce this you can go to the above URL and click on any of the games that show up, or search for a game, or click the Login/Register links.

I'm not sure how important this config line is, but I've tried changing `APP_URL` in the .env file from localhost to `things.ga` and `things.ga/public`.

If you're wondering how my current HTML folder is set up, I have cloned Daze directly into the root of my public HTML folder, just to rule out any issues with subfolders/subdomains. I'm not sure if this application utilizes it, but just in case, I do have mod_rewrite loaded in PHP and .htaccess files enabled in my Apache config.

**Comment by ivankayzer** [comment]
Hey @dinosaw!

Looks like the public folder is not the root of your project. Try adding this to .htaccess in project folder

```
<IfModule mod_rewrite.c>
   RewriteEngine On 
   RewriteRule ^(.*)$ public/$1 [L]
</IfModule>
```

**Comment by dinosaw** [comment]
That worked, thanks! I understand now the concept on how all that worked, basically everything other than the public directory was used to build the resources for the public directory to use. Once I realized that, it made sense and yeah redirecting to the public directory works.

I hate to say it but I'm having another problem. Feel free to just stop helping at any point, hah. I'm sure this is getting old. But basically right now I went ahead and grabbed my own API key so I wouldn't be using yours and that seems to be working fine. Searching the game database from my site brings up results. Also the site itself is working because I'm able to login and register a new user.

But when I search for a game and click on the game page, nothing loads - just a white page. Example, this is the No Man's Sky page:

    http://things.ga/game/3225

Any ideas? APP_DEBUG is set to true and its level to debug. And I'm getting no logs in my Apache error logs this time so I don't have much to go off of.

**Comment by ivankayzer** [comment]
Hey there, @dinosaw!

Files with genres, modes conversion were missing from the project :) 
Pull the latest commit, this should solve your problem!

Thank you very much for being so patient with my broken project, I am really happy to help you.

**Comment by dinosaw** [comment]
Thank you very much for your patience and assistance! I'm really looking forward to getting a project like this build so I can start using it. Keeping up with all my games is a pain, and all the "collector" websites are way too bloated.

Pulling the latest commit did help some, and we're closer, but I'm still running into problems. Now when I go to some games on the main page, or search for games and go to their game page, the page loads fine. But other times I go to the game page and I either get an error page, or I get a blank white page again. Here are three examples, but I ran into quite a few:

    Doom 64 (ErrorException): http://things.ga/game/3471
    Bully (Blank Page): http://things.ga/game/6574
    Chrono Trigger (ErrorException): http://things.ga/game/1802

I still have the dev site up so you can reproduce what I'm talking about

    http://things.ga

Let me know if you have any ideas! No rush.

**Comment by ivankayzer** [comment]
Hey there, @dinosaw!

If you are still interested in this project the errors were fixed.

Thank you for a feedback!

**Comment by dinosaw** [comment]
I'm still interested in this project! For now I've started using player.me as an alternative however - it's pretty decent. But basically I started from scratch again and cloned the most recent repo and I'm now getting this error

```
Exception in random.php line 185:
There is no suitable CSPRNG installed on your system
```

http://i.imgur.com/SUIRSuz.png

**Comment by dinosaw** [comment]
I noticed some recent activity going on with this which I'm excited to see! However I attempted to re-install Daze, but when I get to the step to `npm install`, it installs everything and never prompts me for what it used to (extend my current settings, automatic, project folder, etc). So after running `npm install` once, I ran it again and this was the only output - this could be possibly the issue?

```
[root@server public]# npm install
npm WARN lifecycle undefined~postinstall: cannot run in wd %s %s (wd=%s) undefined bower install /home/cpgames/domains/games.crunchprank.net/public_html
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.17: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```

A rough recap of all what I'm doing to install this:

```
git clone https://github.com/ivankayzer/daze.git .
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php composer-setup.php
php composer.phar install
cp .env.example .env
php artisan key:generate
# at this point i edit .env file with db and api info
php artisan migrate
php artisan db:seed
npm install
```

Any ideas why I'm no longer getting the installation prompts after running `npm install`? I'm using `node 6.9.1`.

**Comment by ivankayzer** [comment]
Hey there, @dinosaw!

You don't see this because I am trying to simplify installation process. All dependencies and handled with bower now. I will release an update explaining the whole process very soon!

**Comment by dinosaw** [comment]
Awesome! Well I'll just keep an eye out on the changes and be patient for now :)

**Comment by ivankayzer** [comment]
Hi, @dinosaw!

Check the readme for simplified installation!

**Comment by dinosaw** [comment]
Heyo! Awesome to see the installation process shortened! However I ran into an issue after completing installation.

Here is a screenshot: http://i.imgur.com/I1OACf7.png

`There is no suitable CSPRNG installed on your system`

I attempted to fix it via the solution [provided here](http://stackoverflow.com/a/38624480). But even after starting from scratch and including that in my `composer.json` file, I received the same error.

**Comment by ivankayzer** [comment]
Hey, @dinosaw!

Try "paragonie/random_compat": "1.2.*"

**Comment by dinosaw** [comment]
So I started from scratch again to be on the safe side, but still getting the error regarding `CSPRNG`.

If it's of any help, this is my entire `composer.json` file. Also I should mention I'm running CentOS, Apache 2.4, and Node 4.7.3.

```{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": ["framework", "laravel"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": ">=5.6.4",
        "laravel/framework": "5.3.*",
        "mashape/unirest-php": "^3.0",
        "guzzlehttp/guzzle": "^6.2",
        "dg/rss-php": "^1.2",
        "paragonie/random_compat": "1.2.*"
    },
    "require-dev": {
        "fzaninotto/faker": "~1.4",
        "mockery/mockery": "0.9.*",
        "phpunit/phpunit": "~5.0",
        "symfony/css-selector": "3.1.*",
        "symfony/dom-crawler": "3.1.*"
    },
    "autoload": {
        "classmap": [
            "database"
        ],
	"psr-4": {
            "Daze\\": "app/"
        }
    },
    "autoload-dev": {
        "classmap": [
            "tests/TestCase.php"
        ]
    },
    "scripts": {
        "post-install-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postInstall",
            "php artisan optimize"
        ],
	"config-cmd" : [
            "php -r \"file_exists('.env') || copy('.env.example', '.env');\"",
                        "php artisan key:generate"
        ],
	"post-update-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postUpdate",
            "php artisan optimize"
        ]
    },
    "config": {
        "preferred-install": "dist"
    }
}
```

Sorry for all the headaches I'm probably causing you lol

**Comment by dinosaw** [comment]
I have resolved the `CSPRNG` issue and it was an error on my end. I had `open_basedir` enabled for whatever reason (which I've had problems with in the past). Thankfully I saw [this post](https://github.com/paragonie/random_compat/issues/99#issuecomment-198380310) that reminded me about it.

Afterward, the site loaded, I was able to create a brand new account and login successfully, I can view game pages without problems, and I can also search for games without issue. However now I'm running into another problem hah :x Basically when I go to a game's overview page and click the "Add" button to add it to a collection, it sends me to an error page:

```
ErrorException in compiled.php line 13002:
Undefined offset: 0
```

Since I've made this ticket long enough already, I thought I would just link to a pastebin that shows the full error output: http://pastebin.com/raw/NpsS0nT3

Any ideas/suggestions? Everything else seems to be working fine.
